pkg/upgrade/upgrade.go

```diff
@@ -84,16 +83,43 @@ func OperatorUninstall(cli client.Client, cfg *rest.Config) error {
 			if err := cli.Delete(context.TODO(), &namespace, []client.DeleteOption{}...); err != nil {
 				return fmt.Errorf("error deleting namespace %v: %w", namespace.Name, err)
 			}
-			fmt.Printf("Namespace %s deleted as a part of uninstall.\n", namespace.Name)
+			fmt.Printf("Namespace %s deleted as a part of uninstallation.\n", namespace.Name)
 		}
 	}
 
-	// Wait for all resources to get cleaned up
+	// give enough time for namespace deletion before proceed
 	time.Sleep(10 * time.Second)
 
-	fmt.Printf("All resources deleted as part of uninstall. Removing the operator csv\n")
+	// We can only assume the subscription is using standard names
+	// if user install by creating different named subs, then we will not know the name
+	// we cannot remove CSV before remove subscription because that need SA account
+	operatorNs, err := GetOperatorNamespace()
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Removing operator subscription which in turn will remove installplan\n")
+	subsName := "opendatahub-operator"
+	if platform == deploy.SelfManagedRhods {
+		subsName = "rhods-operator"
+	} else if platform == deploy.ManagedRhods {
+		subsName = "addon-managed-odh"
+	}
+	sub, _ := deploy.SubscriptionExists(cli, operatorNs, subsName)
+	if sub == nil {
+		fmt.Printf("Could not find subscription %s in namespace %s. Maybe you have a different one", subsName, operatorNs)
+	} else {
+		if err := cli.Delete(context.TODO(), sub); err != nil {
+			return fmt.Errorf("error deleting subscription %s: %w", sub.Name, err)
+		}
+	}
 
-	return removeCsv(cli, cfg)
+	fmt.Printf("Removing the operator CSV in turn remove operator deployment\n")
+	time.Sleep(5 * time.Second)
```

Comment:
instead of sleep maybe better with some wait+poll? no need to block reconcile if we are good after e.g. 1s. 

What is the reason to wait here?

Reply:
i thought about the wait, but here is for the uninstasllation, so there should be no other logic needed even for reconcile. right? like the end of this life.
sleep is not necessary, was added when first remove subscription was added after remove csv, but once it has been moved ahead, this should be cleaned up